### PR TITLE
Enables caption download of a specified format

### PR DESCRIPTION
**Pull request by scentless-apprentice**

### CODE REVIEW

**Comment by scentless-apprentice** [comment]
Looks like there's a testing error because of a dependency on https://github.com/NYTimes/amara/pull/3

File: service/service.go [hunk 1]
```diff
@@ -62,5 +62,8 @@ func (s *CaptionsService) Endpoints() map[string]map[string]http.HandlerFunc {
 		"/jobs/{id}/cancel": {
 			"POST": server.JSONToHTTP(s.CancelJob).ServeHTTP,
 		},
+		"/jobs/{id}/download": {
+			"POST": server.JSONToHTTP(s.DownloadCaption).ServeHTTP,
```
Comment (fsouza) [inline service/service.go]: I think this shouldn't be a JSON endpoint. I guess we want to return just the contents of the captions, but when we do ``server.JSONToHTTP``, Gizmo will marshal the interface returned by the handler as JSON, which would probably fail at this point.

That would require changing the signature of DownloadCaption to ``func(r *http.Request, w http.ResponseWriter)`` and just register it as:

```
"POST": s.DownloadCaption,
```

And we should also make sure that ``DownloadCaption`` sets the proper ``Content-Type`` when returning the content (if we use ``JSONToHTTP``, the ``Content-Type`` will be ``application/json``).

File: service/service_test.go [hunk 2]
```diff
@@ -128,9 +128,10 @@ func TestNewCaptionsService(t *testing.T) {
 	assert.NotNil(service.client.Providers)
 	assert.Equal(service.client.Providers, providers)
 
-	assert.Len(service.Endpoints(), 4)
+	assert.Len(service.Endpoints(), 5)
```
Comment (fsouza) [inline service/service_test.go]: I'd delete this line, I don't think the test should care about how many endpoints the service has, only which endpoints x)

File: service/job.go [hunk 3]
```diff
@@ -141,3 +145,38 @@ func (s *CaptionsService) CreateJob(r *http.Request) (int, interface{}, error) {
...
+		requestLogger.WithError(err).Error("Could not unmarshal download request body")
+		w.WriteHeader(http.StatusBadRequest)
+	}
+
+	captionFile, err := s.client.DownloadCaption(id, param.CaptionFormat)
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+	}
+
+	w.Header().Set("Content-Type", fmt.Sprintf("text/%s", param.CaptionFormat))
```
Comment (fsouza) [inline service/job.go]: Can you append ``; charset="utf-8"`` to the Content-Type just to avoid encoding issues?

**Pull request merged**